### PR TITLE
:sparkles:[#1584] get autorisaties URL from OPENZAAK_DOMAIN or site

### LINKED ISSUE #1584: When specifying a Zaken API authorization with permissions linked to an internal zaaktype, deleting the zaaktype causes a 500 error
**General information**
Open Zaak version: 1.11
Rotterdam nr 80
Sentry 352159

**Describe the bug**
When specifying a Zaken API authorization with permissions linked to an internal zaaktype (eg. you check the zaaktypes individually using the 'Selecteer handmatig' option), deleting the zaaktype that is linked to the autorization causes a 500 error.

ZaakType.DoesNotExist
ZaakType matching query does not exist.
openzaak/components/autorisaties/utils.py in _get_related_object at line 30

A workaround is to delete the corresponding Autorisatie (but note that there isn't a FK so the zaaktype delete isn't cascaded to the authorizations)

I'd suggest to ignore autorizations to entities which no longer exist, if only to allow managing the autorizations. Normally zaaktypes aren't deleted, however this is something that can be done via the admin and was encountered during testing

### PR DESCRIPTION
**Pull request by Coperh**
Fixes #1584

### CODE REVIEW

File: src/openzaak/components/autorisaties/forms.py [hunk 1]
```diff
@@ -392,8 +393,11 @@ def save(self, applicatie: Applicatie, request, commit=True):
             autorisaties = []
             for _type in types:
                 data = autorisatie_kwargs.copy()
-                url = request.build_absolute_uri(_type.get_absolute_api_url())
-                data[_field_info["_autorisatie_type_field"]] = url
+
+                # signal uses build_absolute_url to find autorisaties to delete
```
Comment (Coperh) [inline src/openzaak/components/autorisaties/forms.py:397]: Also does not have access to the request

**Comment by Coperh** [comment]
This only matters if the request is sent with a domain other than the expected one.

Assuming the site or openzaak domain is set properly this should work. If not set properly, the deleting signal will not work regardless.

I am not sure if this affects the rewrite host stuff from #1585

**Review by annashamray — COMMENTED** [review]
Isn't it already fixed by https://github.com/open-zaak/open-zaak/pull/1588/? 
@stevenbal What do you think?

**Comment by annashamray** [comment]
@Coperh Could you reproduce the error in the regression test?

**Comment by stevenbal** [comment]
Is there a way to reproduce this bug locally as well?

**Comment by Coperh** [comment]
Idk if my test works, but it became bugged for me locally.

@stevenbal  basically save autorisaties with on `localhost` or `127.0.0.1` and have your `OPENZAAK_DOMAIN` or site domain be different.

That's what I assume the issue was for the original complain.

**Comment by stevenbal** [comment]
> basically save autorisaties with on localhost or 127.0.0.1 and have your OPENZAAK_DOMAIN or site domain be different. Delete a zaaktype with a related autorisatie. It should be deleted automatically.

Ah I see, I tested it locally on main and this branch, and this fixes it indeed :+1:

**Review by stevenbal — APPROVED** [review]

**Pull request merged**